### PR TITLE
document how extensions hook into UI

### PR DESCRIPTION
**Pull request by feloy**
### What does this PR do?

This PR documents at a high level how the extension is "hooked" into Podman Desktop UI.

### What issues does this PR fix or reference?

References #4529

### CODE REVIEW

File: EXTENSIONS.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ import * as extensionApi from '@podman-desktop/api';
 export async function activate(extensionContext: extensionApi.ExtensionContext): Promise<void> {
   // Create a provider with an example name, ID and icon
   const provider = extensionApi.provider.createProvider({
-    name: 'foobar',
+    name: 'FooBar',
```
Comment (benoitf) [inline EXTENSIONS.md:34]: not sure about this change ?

File: EXTENSIONS.md [hunk 2]
```diff
@@ -46,15 +46,15 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
 
 // Deactivate the extension
 export function deactivate(): void {
-  console.log('stopping foobar extension');
+  console.log('stopping FooBar extension');
```
Comment (benoitf) [inline EXTENSIONS.md:49]: ?

File: EXTENSIONS.md [hunk 3]
```diff
@@ -219,7 +239,7 @@ For example if you contribute a property named `podman.binary.path` it will disp
 export async function activate(extensionContext: extensionApi.ExtensionContext): Promise<void> {
   // Define the provider
   const provider = extensionApi.provider.createProvider({
-    name: 'foobar',
+    name: 'FooBar',
```
Comment (benoitf) [inline EXTENSIONS.md:242]: ?

**Review by benoitf — APPROVED** [review]

File: EXTENSIONS.md (hunk 1 above)
Comment (feloy) [reply EXTENSIONS.md:34]: I would like to differentiate the Id and the Name, to make clearer which value is used to reference the provider (the id) and which one is a displayed only value (the name).

Comment (benoitf) [reply EXTENSIONS.md:34]: works for me, I've approved the PR

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**